### PR TITLE
Signup flow

### PR DESCRIPTION
**Pull request by ehfeng**
Changes proposed in this pull request:
- adding "?signup" param when new organizations are created, for on-boarding flow
- also changed <Link> to <a> to ensure reloading of page in the platform.jsx page (hashchange isn't triggered otherwise)

@getsentry/team

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx [hunk 1]
```diff
@@ -142,13 +142,13 @@ const ProjectInstallPlatform = React.createClass({
             <div dangerouslySetInnerHTML={{__html: this.state.html}}/>
           )}
 
-          {queryParams.onboarding ?
+          {'signup' in queryParams ?
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Why not `queryParams.signup` ?

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx [hunk 2]
```diff
@@ -142,13 +142,13 @@ const ProjectInstallPlatform = React.createClass({
...
             <p>
-              <Link
-                to={`/${orgId}/${projectId}/?onboarding=1#welcome`}
+              <a
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Why is this changed from a `<Link />` to an `<a />`?

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx [hunk 3]
```diff
@@ -142,13 +142,13 @@ const ProjectInstallPlatform = React.createClass({
...
+                href={`/${orgId}/${projectId}/#welcome`}
                 className="btn btn-primary btn-lg">
-                  Got it! Take me to the Issue Stream.
-              </Link>
+                  {tct(`Got it! Take me to the Issue Stream.`)}
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Use quotes, not backticks here.

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 1 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Returns empty string, which evals to false. I could do `queryParams.signup != null` instead, if preferred.

File: src/sentry/templates/sentry/create-project.html [hunk 4]
```diff
@@ -27,7 +27,16 @@
       {{ form.team|as_crispy_field }}
     {% endif %}
 
+    {% if 'signup' in request.GET %}
```
Comment (mattrobenolt) [inline src/sentry/templates/sentry/create-project.html]: You don't need to do this. `?signup` will already be in the `GET` for this page, which means the `action` is going to `POST` back to this same url. Inside your view, you could just check for `'signup' in request.GET` again without this extra `POST` field.

File: src/sentry/web/frontend/create_project.py [hunk 5]
```diff
@@ -78,10 +78,15 @@ def handle(self, request, organization):
...
             project = form.save(request.user, request.META['REMOTE_ADDR'])
 
-            return self.redirect(absolute_uri('/{}/{}/settings/install/'.format(
+            install_uri = absolute_uri('/{}/{}/settings/install/'.format(
                 organization.slug,
                 project.slug,
-            )))
+            ))
+
+            if 'signup' in request.POST:
```
Comment (mattrobenolt) [inline src/sentry/web/frontend/create_project.py]: Related to above comment, just need to check `'signup' in request.GET`.

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 2 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: I'm adding a #welcome to the url, which triggers the onboarding panel to be opened. hashchange event isn't being triggered with <Link>, nor is the component being remounted. I think it's something to do with React Router, as they use the hash as a fallback.

Changing it to <a /> reloads the whole page, which allows me to open the on-boarding panel onComponentMount.

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 1 above)
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Oh, in that case, do `queryParams.hasOwnProperty('signup')`. It's more reliable than `'signup' in queryParams` for JavaScript.

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 2 above)
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Uhhhh, this is out of my realm of understanding.

/cc @macqueen @benvinegar

File: src/sentry/web/frontend/create_project.py (hunk 5 above)
Comment (ehfeng) [inline src/sentry/web/frontend/create_project.py]: fixed

File: src/sentry/templates/sentry/create-project.html (hunk 4 above)
Comment (ehfeng) [inline src/sentry/templates/sentry/create-project.html]: fixed

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 3 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: fixed

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 1 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: fixed

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 2 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: this is a nice-to-have. Getting the signup stuff out is more important to me in the short term, so I can just change this back to <Link>. Or @macqueen if you have any ideas on how to propagate the state change, happy to do that.

File: src/sentry/static/sentry/app/views/projectInstall/overview.jsx [hunk 6]
```diff
@@ -17,11 +17,11 @@ const ProjectInstallOverview = React.createClass({
 
   getIntegrationLink(root, platform, display) {
     let {orgId, projectId} = this.props.params;
-    let onboarding = this.props.location.query.onboarding ? '?onboarding=1' : '';
+    let signup = 'signup' in this.props.location.query ? '?signup' : '';
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/overview.jsx]: Use `hasOwnProperty` check here too.

File: src/sentry/templates/sentry/create-project.html [hunk 7]
```diff
@@ -30,4 +30,9 @@
     <button type="submit" class="btn btn-primary" name="action_save">{% trans "Create Project" %}</button>
   </form>
 </div>
+
+{% block post_content %}{% endblock %}
+
 {% endblock %}
+
```
Comment (mattrobenolt) [inline src/sentry/templates/sentry/create-project.html]: Get rid of this trailing whitespace.

Comment (ehfeng) [inline src/sentry/templates/sentry/create-project.html]: fixed

File: src/sentry/static/sentry/app/views/projectInstall/overview.jsx (hunk 6 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/overview.jsx]: fixed

File: src/sentry/templates/sentry/create-project.html [hunk 8]
```diff
@@ -30,4 +30,7 @@
     <button type="submit" class="btn btn-primary" name="action_save">{% trans "Create Project" %}</button>
   </form>
 </div>
+
```
Comment (mattrobenolt) [inline src/sentry/templates/sentry/create-project.html]: None of this changed whitespace is even needed in this file now.

File: src/sentry/templates/sentry/create-project.html [hunk 9]
```diff
@@ -30,4 +30,4 @@
     <button type="submit" class="btn btn-primary" name="action_save">{% trans "Create Project" %}</button>
   </form>
 </div>
-{% endblock %}
+{% endblock %}
```
Comment (mattrobenolt) [inline src/sentry/templates/sentry/create-project.html]: lol

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 2 above)
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: I would at least write a comment TODO that this was done intentionally.

Comment (benvinegar) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Otherwise someone will come along and go, "this is dumb" and re-write it to use `<Link>`.

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx [hunk 10]
```diff
@@ -142,13 +142,13 @@ const ProjectInstallPlatform = React.createClass({
             <div dangerouslySetInnerHTML={{__html: this.state.html}}/>
           )}
 
-          {queryParams.onboarding ?
+          {queryParams.hasOwnProperty('signup') ?
             <p>
-              <Link
-                to={`/${orgId}/${projectId}/?onboarding=1#welcome`}
+              <a
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: Do what @benvinegar said. Add a `NOTE` or `TODO` comment here to explain so someone else doesn't change it.

File: src/sentry/static/sentry/app/views/projectInstall/platform.jsx (hunk 2 above)
Comment (ehfeng) [inline src/sentry/static/sentry/app/views/projectInstall/platform.jsx]: fixed

**Comment by mattrobenolt** [comment]
:cool:
